### PR TITLE
[code-completion] Suggest module names, and handle module name bases for member completion in type position

### PR DESCRIPTION
**Pull request by nathawes**
Previous we didn't suggest modules for top-level completions, and for module type member completions we would just return if the parsed type loc didn't type-check as a type.

Resolves rdar://problem/56800640
Resolved rdar://problem/56801340

### CODE REVIEW

**Comment by nathawes** [comment]
@swift-ci please test

File: test/IDE/complete_from_swift_module.swift [hunk 1]
```diff
@@ -99,3 +105,20 @@ func testPostfixOperator1(x: Int) {
 // TOP_LEVEL_1-DAG: Decl[GlobalVar]/Local:     hiddenImport[#Int#]{{; name=.+$}}
 // TOP_LEVEL_1-DAG: Decl[GlobalVar]/OtherModule[foo_swift_module]:     globalVar[#Int#]{{; name=.+$}}
 // TOP_LEVEL_1: End completions
+
+struct Foo: Swift.Array.#^STDLIB_TYPE_QUALIFIED^# {}
+// STDLIB_TYPE_QUALIFIED: Begin completions
+// STDLIB_TYPE_QUALIFIED: Decl[TypeAlias]/CurrNominal: Index[#Int#]; name=Index
+// STDLIB_TYPE_QUALIFIED: Decl[TypeAlias]/CurrNominal: Element[#Element#]; name=Element
+// STDLIB_TYPE_QUALIFIED: Keyword/None: Type[#Array.Type#]; name=Type
+// STDLIB_TYPE_QUALIFIED: End completions
```
Comment (rintaro) [inline test/IDE/complete_from_swift_module.swift]: I believe this test case works even before this change.
Could you add a test for `Swift.#^TOKEN^#` at type position as well?

**Review by rintaro — CHANGES_REQUESTED** [review]

File: test/IDE/complete_from_swift_module.swift (hunk 1 above)
Comment (nathawes) [reply test/IDE/complete_from_swift_module.swift]: Yeah, it was already working. Added the extra test.

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@swift-ci please test

File: test/SourceKit/CodeComplete/complete_sort_order.swift [hunk 2]
```diff
@@ -33,6 +33,8 @@ func test() {
 // CONTEXT: key.kind: source.lang.swift.decl
 // CONTEXT-NEXT: key.name: "x"
 // CONTEXT-NOT: key.name:
+// CONTEXT: key.name: "complete_sort_order",
+// CONTEXT-NOT: key.name:
 // CONTEXT: key.name: "foo(a:)"
 // CONTEXT-NOT: key.name:
 // CONTEXT: key.name: "foo(a:)"
```
Comment (rintaro) [inline test/SourceKit/CodeComplete/complete_sort_order.swift]: `complete_sort_order` is the module name of the current file. It's a little strange if it precedes other decls in the module.
It probably makes sense if `Decl[Module]` has lower priority than other decls.
@benlangmuir what do you think?

Comment (benlangmuir) [reply test/SourceKit/CodeComplete/complete_sort_order.swift]: I agree

**Comment by nathawes** [comment]
@swift-ci please test

**Review by rintaro — APPROVED** [review]
LGTM!

**Pull request merged**